extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -297,6 +303,7 @@ export abstract class ToolBase implements ITool {
 		return !version || (this._version ? SemVerCompare(this._version, version) >= 0 : false);
 	}
 
+	private _pendingVersionAndStatusUpdate: Promise<void> = Promise.reject();
```

Comment:
Just leave this as undefined to start with - since you set it in the constructor you're guaranteed to have it by the time anyone could call it and having the default value be reject is weird